src/django_elasticsearch_dsl_drf/pagination.py

```diff
@@ -53,12 +53,11 @@ def page(self, number):
         :param number:
         :return:
         """
-        number = self.validate_number(number)
         bottom = (number - 1) * self.per_page
         top = bottom + self.per_page
-        if top + self.orphans >= self.count:
```

Comment:
In any case, this shouldn't have been removed.

```python
if top + self.orphans >= self.count:
    top = self.count
```

Reply:
I think it should be removed or rewritten. The `self.count` raises the unwanted ES hit.
Regarding my testing, it works just fine - the ES is inteligent enough not to cause problems when the `top` is higher than element count.

But maybe I have missed something - i.e. I don't know, what is `self.orphans`.

Reply:
OK, I see the failed tests in Travis, so I will take look at them.

Reply:
I am still not sure, what is the problem with the failing tests - I expect, that the problem is, that it gets different ES data structure than on my local environment.

I was not able to run the tests locally, but I will try further tomorrow.

Reply:
Regarding removal of orphans.

[The minimum number of items allowed on the last page, defaults to zero. Use this when you don’t want to have a last page with very few items. If the last page would normally have a number of items less than or equal to orphans, then those items will be added to the previous page (which becomes the last page) instead of leaving the items on a page by themselves. For example, with 23 items, per_page=10, and orphans=3, there will be two pages; the first page with 10 items and the second (and last) page with 13 items.](https://github.com/django/django/blob/master/django/core/paginator.py#L79)

In general, although it's a technically good optimisation fix for Elasticsearch, it does break some commonly accepted functionality. Additionally, page number validation is messed up with.

I fixed error in your branch [here](https://github.com/barseghyanartur/django-elasticsearch-dsl-drf/commit/5f92add7d31fee35140559def62b8fb7a581bd6e).

I would accept additional paginator classes though, but not as defaults. Let them be `QueryFriendlyPaginator`, `QueryFriendlyPageNumberPagination` and `QueryFriendlyLimitOffsetPagination`. Probably `validate_number` shall be overwritten for these classes too (to avoid the count query).

I'll leave this up to you.

Reply:
@barseghyanartur Thank you for the fix.

Regarding the orphans - I would fix this behaviour by requesting the orphaned nodes in additional request. So the number of ES hits will be 1 in most cases, but with if there are orphans, it would fall back to 2 requests. That way it would be beneficial for most users.
Do you think it is a good solution?